jsons/serializers/default_object.py

```diff
@@ -40,69 +48,189 @@ def default_object_serializer(
+        '_store_cls': Verbosity.WITH_CLASS_INFO in verbose
+    }
+
+    result = _do_serialize(obj=obj,
+                           cls=cls,
+                           attributes=attributes,
+                           kwargs=kwargs_,
+                           key_transformer=key_transformer,
+                           strip_nulls=strip_nulls,
+                           strip_privates=strip_privates,
+                           strip_properties=strip_properties,
+                           strip_class_variables=strip_class_variables,
+                           strip_attr=strip_attr,
+                           strict=strict,
+                           fork_inst=fork_inst)
+
+    cls_name = get_class_name(cls, fully_qualified=True, fork_inst=fork_inst)
     if not kwargs.get('_store_cls'):
-        result = _get_dict_with_meta(result, cls_name, verbose,
-                                     kwargs['fork_inst'])
+        result = _get_dict_with_meta(result, cls_name, verbose, fork_inst)
+    return result
+
+
+def _do_serialize(
+        obj: object,
+        cls: type,
+        attributes: Dict[str, Optional[type]],
+        kwargs: dict,
+        key_transformer: Optional[Callable[[str], str]] = None,
+        strip_nulls: bool = False,
+        strip_privates: bool = False,
+        strip_properties: bool = False,
+        strip_class_variables: bool = False,
+        strip_attr: Union[str, MutableSequence[str], Tuple[str]] = None,
+        strict: bool = False,
+        fork_inst: Optional[type] = StateHolder) -> Dict[str, object]:
+    result = dict()
+    for attr_name, cls_ in attributes.items():
+        attr = obj.__getattribute__(attr_name)
+        dumped_elem = None
+        try:
+            dumped_elem = dump(attr,
+                               cls=cls_,
+                               key_transformer=key_transformer,
+                               strip_nulls=strip_nulls,
+                               strip_privates=strip_privates,
+                               strip_properties=strip_properties,
+                               strip_class_variables=strip_class_variables,
+                               strip_attr=strip_attr,
+                               **kwargs)
+
+            _store_cls_info(dumped_elem, attr, kwargs)
+        except RecursionDetectedError:
+            fork_inst._warn('Recursive structure detected in attribute "{}" '
+                            'of object of type "{}", ignoring the attribute.'
+                            .format(attr_name, get_class_name(cls)))
+        except SerializationError as err:
+            if strict:
+                raise
+            else:
+                fork_inst._warn('Failed to dump attribute "{}" of object of '
+                                'type "{}". Reason: {}. Ignoring the '
+                                'attribute.'
+                                .format(attr, get_class_name(cls), err.message))
+                break
```

Comment:
Is there a case where you want an object to be half-serialized? At least, I would expect it to fail by default. In the current version, `strict=False` by default.

Reply:
I wonder if more people would prefer the strict behavior as default.